src/palace/manager/api/controller/work.py

```diff
@@ -96,7 +98,9 @@ def contributor(
             max_age=lane.max_cache_age(), mime_types=flask.request.accept_mimetypes
         )
 
-    def permalink(self, identifier_type, identifier):
+    def permalink(
+        self, identifier_type: str, identifier: str
+    ) -> OPDSEntryResponse | ProblemDetail:
```

Comment:
Added some type hinting here so mypy checks this function, which would have given us an indication this could have been a problem.